src/osp/scientific/kepler.h

```diff
@@ -0,0 +1,161 @@
+        double m_E0;
+    };
+
+    // Gravitational parameter of the central body (G*M)
+    double m_gravitationalParameter;
+
+    KeplerOrbit(double semiMajorAxis,
+                double eccentricity,
+                double inclination,
+                double argumentOfPeriapsis,
+                double longitudeOfAscendingNode,
+                double meanAnomalyAtEpoch,
+                double epoch,
+                double gravitationalParameter,
+                double r0,
+                double v0,
+                double m_E0,
+                double h) : m_semiMajorAxis(semiMajorAxis),
+                            m_eccentricity(eccentricity),
+                            m_inclination(inclination),
+                            m_argumentOfPeriapsis(argumentOfPeriapsis),
+                            m_longitudeOfAscendingNode(longitudeOfAscendingNode),
+                            m_meanAnomalyAtEpoch(meanAnomalyAtEpoch),
+                            m_epoch(epoch),
+                            m_gravitationalParameter(gravitationalParameter),
+                            m_r0(r0),
+                            m_v0(v0),
+                            m_E0(m_E0),
+                            m_h(h) {}
+
+    /**
+     * @brief Get the eccentric anomaly at a given time
+     */
+    double get_eccentric_anomaly(double time) const;
+
+    /**
+     * @brief Get the true anomaly at a given eccentric anomaly
+     *
+     * @note This is not a very satisfying interface for this function,
+     * but having a "get_true_anomaly_at_time" would require double the computation if you wanted both
+     */
+    double get_true_anomaly_from_eccentric(double eccentric_anomaly) const;
+
+    /**
+     * @brief Compute the position, velocity of the orbit at a given eccentric anomaly
+     */
+    void get_state_vectors_at_eccentric_anomaly(double true_anomaly, Vector3d &position, Vector3d &velocity) const;
+
+public:
+    /**
+     * @brief Compute the position, velocity of the orbit at a given time
+     */
+    void get_state_vectors_at_time(double time, Vector3d &position, Vector3d &velocity) const;
+
+    /**
+     * @brief Compute the acceleration of the orbit at a given position
+     */
+    Vector3d get_acceleration(const Vector3d &radius) const;
+
+    /**
+     * @brief Returns a fixed reference frame for the orbit, invariant of time
+     */
+    void get_fixed_reference_frame(Vector3d &radial, Vector3d &transverse, Vector3d &outOfPlane) const;
+
+    /**
+     * @brief Find Apoapsis of orbit
+     *
+     * @return std::optional<double> Apoapsis distance, if eccentricity < 1
+     */
+    std::optional<double> get_apoapsis() const;
+
+    /**
+     * @brief Find period of orbit if elliptic
+     */
+    std::optional<double> get_period() const;
+
+    /**
+     * @brief Find Periapsis of orbit
+     */
+    double get_periapsis() const;
+
+    inline bool is_elliptic() const { return m_eccentricity < 1.0 - KINDA_SMALL_NUMBER; }
```

Comment:
inline keyword isn't needed on functions defined inside the class body like this.

better would be `constexpr` and `noexcept`

Reply:
some extra context: function bodies inside classes/structs are automatically considered inline